src/navigation/OnboardingStack.tsx

```diff
@@ -1,62 +1,154 @@
-    <Stack.Screen
-      name={OnboardingScreens.PersonalPrivacy}
-      component={PersonalPrivacy}
-    />
-    <Stack.Screen
-      name={OnboardingScreens.NotificationDetails}
-      component={NotificationDetails}
-    />
-    <Stack.Screen
-      name={OnboardingScreens.ShareDiagnosis}
-      component={ShareDiagnosis}
-    />
-    <Stack.Screen
-      name={OnboardingScreens.NotificationPermissions}
-      component={NotificationPermissions}
-    />
-    <Stack.Screen
-      name={OnboardingScreens.EnableExposureNotifications}
-      component={EnableExposureNotifications}
-    />
-    <Stack.Screen
-      name={OnboardingScreens.LanguageSelection}
-      component={LanguageSelection}
-    />
-  </Stack.Navigator>
-)
+const OnboardingStack: FunctionComponent = () => {
+  const { t } = useTranslation()
+  const navigation = useNavigation()
+
+  const onboardingScreenOptions: StackNavigationOptions = {
+    headerShown: false,
+  }
+
+  const HeaderRight = (stepNumber: number) => {
+    const determineStepText = () => {
+      switch (stepNumber) {
+        case 1:
+          return t("onboarding.step_1_of_3")
+        case 2:
+          return t("onboarding.step_2_of_3")
+        case 3:
+          return t("onboarding.step_3_of_3")
+        default:
+          return t("onboarding.step_1_of_3")
+      }
+    }
+
+    return (
+      <View style={style.headerRight}>
+        <GlobalText style={style.headerRightText}>
+          {determineStepText()}
+        </GlobalText>
+        <TouchableOpacity
+          onPress={() =>
+            navigation.navigate(OnboardingScreens.ValueProposition)
+          }
+        >
+          <SvgXml
+            xml={Icons.Close}
+            fill={Colors.darkestGray}
+            style={style.closeIcon}
+          />
+        </TouchableOpacity>
+      </View>
+    )
+  }
+
+  const appSetupOptions: StackNavigationOptions = {
+    headerShown: true,
+    headerLeft: () => null,
+    headerTitleAlign: "left",
+    headerTitle: "App Setup",
+    headerTitleStyle: style.headerTitle,
+  }
+
+  return (
+    <Stack.Navigator screenOptions={onboardingScreenOptions}>
+      <Stack.Screen name={OnboardingScreens.Welcome} component={Welcome} />
+      <Stack.Screen
+        name={OnboardingScreens.EulaModal}
+        component={EulaModal}
+        options={{ ...TransitionPresets.ModalTransition }}
+      />
+      <Stack.Screen
+        name={OnboardingScreens.Introduction}
+        component={Introduction}
+      />
+      <Stack.Screen
+        name={OnboardingScreens.PhoneRemembersDevices}
+        component={PhoneRemembersDevices}
+      />
+      <Stack.Screen
+        name={OnboardingScreens.PersonalPrivacy}
+        component={PersonalPrivacy}
+      />
+      <Stack.Screen
+        name={OnboardingScreens.GetNotified}
+        component={GetNotified}
+      />
+      <Stack.Screen
+        name={OnboardingScreens.ValueProposition}
+        component={ValueProposition}
+      />
+      <Stack.Screen
```

Comment:
Curious of anyone's thoughts re: these two screens, and eventually the `EulaModal` as well, living in their own "App Setup" stack.

Reply:
That would make sense to me. They are pretty distinct functionally from the onboarding info screens, so I would lean towards them being grouped separately.

Reply:
Could I get your thoughts on 288893b105d9c4e998dc1d40acdcba9ffeba6db4?